### PR TITLE
docs: fix some link errors

### LINKED ISSUE #656: :open_book: `Docs`: Some links should be updated
## 1. 404
image

## 2. render error
image
image

## 3. 404
image

## 4. same as No.2
image

## 5. similar to No.2
image

### PR DESCRIPTION
**Pull request by aFlyBird0**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](../CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description
Please see #656 for more information.

## Related Issues
close #656

### CODE REVIEW

File: docs/commands/verify.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ If not, it tries to give you a hint that maybe you forgot to run `dtm init` firs
 
 ## 4 Config / State / Resource
 
-For definitions of _Config_, _State_, and _Resource_, see [Core Concepts](../core-concepts/core-concepts).
+For definitions of _Config_, _State_, and _Resource_, see [Core Concepts](../core-concepts/core-concepts.md).
```
Comment (daniel-hutao) [inline docs/commands/verify.md:26]: `.md` is no needed

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

File: docs/commands/verify.md (hunk 1 above)
Comment (aFlyBird0) [reply docs/commands/verify.md:26]: see here, it doesn't end with `.md` or `/`, but [the coresponding link](https://docs.devstream.io/en/latest/commands/core-concepts/core-concepts) is fail to render.
https://github.com/devstream-io/devstream/blob/90d0a07ab57b34d60ed8e8c59bf685f2d93d99b9/docs/commands/verify.md?plain=1#L26

So, I think I have to build docs with `mkdocs` in local. But I get the error `Config file 'mkdocs.yml' does not exist.`.

So, how should we build docs of dtm? I need your help, @daniel-hutao , thanks. And maybe we talk about it tommorrow, just go to hava a break now.

Comment (daniel-hutao) [reply docs/commands/verify.md:26]: image

image

`mkdocs` can be used locally. Please ensure that the execution path is in the project root directory. If there are still errors, please let me know again.

Comment (aFlyBird0) [reply docs/commands/verify.md:26]: Sorry, I entered wrong path(`/docs`). Thank you. I will retry tomorrow, Good Night!

**Comment by aFlyBird0** [comment]
If we we want to ref `core-concepts.md` in `verfiy.md`:

image

After test, there are three availiable ways:

1. relative path: `../../core-concepts/core-concepts`. It seems that mkdocs regard each file as a page(dir), so we need `../../` instead of `../`
2. absolute path: `/core-concepts/core-concepts`. (with the `docs directory as the root directory`)
3. using md: `../core-concepts/core-concepts.md` (this is relative path, absolute path is similar).

I prefer the second one.

This comment can be used as a reference for future document reference changes. @daniel-hutao

**Comment by aFlyBird0** [comment]
> If we we want to ref `core-concepts.md` in `verfiy.md`:
> 
> image
> 
> After test, there are three availiable ways:
> 
> 1. relative path: `../../core-concepts/core-concepts`. It seems that mkdocs regard each file as a page(dir), so we need `../../` instead of `../`
> 2. absolute path: `/core-concepts/core-concepts`. (with the `docs` directory as the root directory)
> 3. using md: `../core-concepts/core-concepts.md` (this is relative path, absolute path is similar).
> 
> **I prefer the second one.**
> 
> This comment can be used as a reference for future document reference changes. @daniel-hutao

But the third way is supported to read in local with markdown reader.

**Comment by aFlyBird0** [comment]
> If we we want to ref `core-concepts.md` in `verfiy.md`:
> 
> image
> 
> After test, there are three availiable ways:
> 
> 1. relative path: `../../core-concepts/core-concepts`. It seems that mkdocs regard each file as a page(dir), so we need `../../` instead of `../`
> 2. absolute path: `/core-concepts/core-concepts`. (with the `docs` directory as the root directory)
> 3. using md: `../core-concepts/core-concepts.md` (this is relative path, absolute path is similar).
> 
> **I prefer the second one.** But the third way is supported to read in local with markdown reader.
> 
> This comment can be used as a reference for future document reference changes.

I need your opinion, @daniel-hutao

**Comment by daniel-hutao** [comment]
@aFlyBird0 I agree with your choice, 2 is better.

**Comment by aFlyBird0** [comment]
I switch style of all links except `plugins` dir into `way 2`. And I haved checked each edited links in local. Please Review @daniel-hutao

**Pull request merged**